internal/lib/restore.go

```diff
@@ -63,8 +64,44 @@ func (c *ContainerServer) ContainerRestore(ctx context.Context, opts *ContainerC
 	}
 
 	if ctr.RestoreArchive() != "" {
-		if err := crutils.CRImportCheckpointWithoutConfig(ctr.Dir(), ctr.RestoreArchive()); err != nil {
-			return "", err
+		if ctr.RestoreIsOCIImage() {
+			logrus.Debugf("Restoring from %v\n", ctr.RestoreArchive())
+			imageMountPoint, err := c.StorageImageServer().GetStore().MountImage(ctr.RestoreArchive(), nil, "")
+			if err != nil {
+				return "", err
+			}
+			logrus.Debugf("Checkpoint image mounted at %v\n", imageMountPoint)
+			defer func() {
+				_, err := c.StorageImageServer().GetStore().UnmountImage(ctr.RestoreArchive(), true)
+				if err != nil {
+					logrus.Errorf("Failed to unmount checkpoint image: %q", err)
+				}
+			}()
+
+			// Import all checkpoint files except ConfigDumpFile and SpecDumpFile. We
+			// generate new container config files to enable to specifying a new
+			// container name.
+			checkpoint := []string{
+				"artifacts",
+				metadata.CheckpointDirectory,
+				// metadata.CheckpointVolumesDirectory,
```

Comment:
Is this intentional? Do we wanna left a TODO if so?

Reply:
That was a leftover. I removed it.